src/main/resources/META-INF/rewrite/jspecify.yml

```diff
@@ -0,0 +1,69 @@
+#
+# Copyright 2024 the original author or authors.
+# <p>
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# <p>
+# https://www.apache.org/licenses/LICENSE-2.0
+# <p>
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.jspecify.MigrateToJspecify
+displayName: Migrate to jspecify
+description: >-
+  This recipe will migrate to jspecify annotations.
+tags:
+  - java
+recipeList:
+  - org.openrewrite.java.jspecify.MigrateFromJavaxAnnotationApi
+  - org.openrewrite.java.jspecify.MigrateFromJakartaAnnotationApi
+
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.jspecify.MigrateFromJavaxAnnotationApi
+displayName: Migrate to JSpecify
+description: Migrate from javax Annotation API to JSpecify.
+recipeList:
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: javax.annotation.Nullable
+      newFullyQualifiedTypeName: org.jspecify.annotations.Nullable
+      ignoreDefinition: true
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: javax.annotation.Nonnull
+      newFullyQualifiedTypeName: org.jspecify.annotations.NonNull
+      ignoreDefinition: true
+  - org.openrewrite.java.dependencies.AddDependency:
+      groupId: org.jspecify
+      artifactId: jspecify
+      version: latest.release
```

Comment:
Might be best to add an `onlyIfUsing` expression here, such that we only add the dependency where needed.

Reply:
I just added `onlyIfUsing` but it doesn't seem to be working.  

I tried this:
```
 onlyIfUsing: org.jspecify.annotations..*
```

I also tried this:
```
 onlyIfUsing: org.jspecify.annotations.*
```

Reply:
There's a bit of an implementation detail here with how that `onlyIfUsing` is implemented, and how scanning recipes work. Scanning recipes run a first separate scan across all source files, before making informed changes downstream. We only do a single scan across all files, as otherwise we could potentially have to do a lot of passes through very large projects, slowing things down considerably. The `AddDependency` recipe is a scanning recipe, that evaluates the `onlyIfUsing` in that initial pass through, seeing the files as they are before the `ChangeType` recipes make their changes. That means that if we evaluate `org.jspecify.annotations.*`, then it's not (yet) seeing those annotations being used, so it's not adding a dependency, even if effectively in the end we do end up using those annotations. The trick then is to look at the "before" situation instead, looking for each of the old annotations, and adding the JSpecify dependency if found, as we expect those annotations to be converted in subsequent steps. Hope that helps explain the issue!